src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -105,7 +105,7 @@ public function register_routes() {
 				// Excludes invalid directory name characters: `/:<>*?"|`.
 				'([^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?)',
 				// Matches the template name.
-				'[\/\w-]+'
+				'[\/\pL0-9_-]+'
```

Comment:
@kozer What's the significance of this regex change?

Reply:
@danielbachhuber let me explain:

Before it was `\w`: This actually matches the following: `a-z`, A-Z`,  `0-9` and `_`.
Changing this to `\pL0-9_`, we match the same as before, plus all the unicode characters of all the languages mentioned [here](https://www.php.net/regexp.reference.unicode).

I also kept the `0-9_` to match what we had before with `\w`.

Reply:
@kozer Cool! Can we update the comment immediately above it to communicate as such? I think `\w` is more commonly used (for better or for worse) so it would be good to have a note as to why `\w` isn't used.